ProjectLighthouse.Servers.GameServer/Controllers/MessageController.cs

```diff
@@ -66,12 +69,39 @@ public async Task<IActionResult> Announce()
                                   $"token.ExpiresAt: {token.ExpiresAt.ToString()}\n" +
                                   "---DEBUG INFO---");
         #endif
-        
+
         return this.Ok(announceText.ToString());
     }
 
     [HttpGet("notification")]
-    public IActionResult Notification() => this.Ok();
+    [Produces("text/xml")]
+    public async Task<IActionResult> Notification()
+    {
+        GameTokenEntity token = this.GetToken();
+
+        IQueryable<NotificationEntity> notificationQuery = this.database.Notifications
+            .Where(n => n.UserId == token.UserId)
+            .Where(n => n.IsDismissed == false)
+            .OrderByDescending(n => n.Id);
+
+        // We don't need to do any more work if there are no unconverted notifications to begin with.
+        if (!await notificationQuery.AnyAsync()) return this.Ok();
```

Comment:
I think this should be simplified to have one query that gets a list of your notifications then we can mark them as dismissed locally then save so we only have 2 database round trips because right now it makes 3. There's one for the count, one for the actual data, and one for the update.